sklearn/preprocessing/data.py

```diff
@@ -323,7 +323,7 @@ def transform(self, X):
         """
         check_is_fitted(self, 'scale_')
 
-        X = check_array(X, copy=self.copy, ensure_2d=False)
+        X = check_array(X, copy=self.copy, ensure_2d=False, dtype=FLOAT_DTYPES)
```

Comment:
> :confused: is that not going to cast a int32 dtype into a float64:

Correct, but at least it won't cast a float32 into a float64, which is
the most important thing to avoid.